### DISCUSSION TITLE
Better Documentation

### DISCUSSION THREAD

**Discussion by antistic** [opensteno/plover#1370] [General]
With Plover getting more attention (look how many stars we have now!!), it's probably time to start improving all the documentation.

## Current State

- [OpenStenoProject.org](https://www.openstenoproject.org/)
  - A pretty simple site explaining the project, what stenography is, and links to the beginner's guide on the wiki
- [Plover Homepage](https://www.openstenoproject.org/plover/)
  - The Plover Homepage, which is kind of hard to get to (there's one in-paragraph link from the open steno homepage)
  - Seems to get linked more than the open steno homepage
- [Github Wiki](https://github.com/openstenoproject/plover/wiki)
  - Has lots of user-contributed information, including the beginner's guide, installation guide, FAQ, open steno history...
- [Read The Docs](https://plover.readthedocs.io/en/latest/)
  - Mostly developer-oriented documentation.

## Problems

 - Lots of information overlap between sources, such as what hardware to use, what stenography is, and where to learn.
     - This means that it's really hard to keep up to date and consistent. For example, "Art of Chording" is only linked in some of the learning resources pages.
     - It makes it hard to know where to look for the best information
 - It's not clear which resource is for what purpose. The wiki has kind of become a catch all.
 - People are not getting the information they need when they need it. Many people come to the discord with no idea where to start or what's going on.

## Solutions

I'd love to get everyone's thoughts on how we should solve this.

A tentative idea:
 - The open steno homepage should be the place for marketing and advertising steno: explaining what steno is, why people should use it, etc.
 - The plover homepage should be an installation guide and a beginner's guide
 - The Read The Docs should be for developers (both for plover and plover plugin developers)
 - I'm not sure about the wiki--I love all the user contributed content there, but the line between some information on the wiki, discord, and github discussions is very blurry. Having a place where any user can contribute ideas (without the daunting experience of needing to do a pull request) is very important, but can lead to confusing, unstructured content.

**Comment by antistic** [comment]
Various things that have come up on the discord:

1. Wiki has too many pages. e.g. the mnemonics page does not fit, [this](https://github.com/openstenoproject/plover/wiki/Converting-a-normal-keyboard-(staggered-NKRO-QWERTY)-to-a-dedicated-Plover-keyboard) and the stentura guide should go under hardware
1. [this](https://github.com/openstenoproject/plover/wiki/Main-dictionary-changes) should be in the changelog
1. There are a few useful pages related to dictionary and all that, like the brief ideas page or the misleading strokes page
1. Does the [top 100 english words](https://github.com/openstenoproject/plover/wiki/Top-100-English-Words-in-Plover%27s-Dictionary) belong on the wiki? It's more of a learning resource, and maybe belongs more in something like Art of Chording
1. Should the wiki be an intro to steno, or documentation for Plover?
1. The wiki maybe shouldn't be a place to convince people to steno -- this should be on open steno
1. What do we want as the ideal landing page for someone who saw mirabai talk? The open steno site?
1. The ideal getting started guide should not assume the reader knows what steno is
1. We should run some user studies/ask what kind of information the wider community wants
1. We should have a community run learning resource, like AoC (but is AoC the right fit? It only deals with theory, not any of the technical side)
1. We should distinguish between plover the theory and plover the program
1. The dictionary format page is too dense for new people
1. When someone wants to add to the dictionary, they need to know:
  1. How to use the software
  1. How to make a good entry (theory)
  1. Software workflows to know which dictionary to add it to
1. You can't easily go to the open steno homepage from the plover homepage
1. Everything should be discoverable from the OSP homepage
1. The OSP homepage should be split into the different sections: software, hardware and learning resources
1. Community resources (plugins, theories etc.) should be linked. (however, putting them on the main site could have problems, since community stuff pops up all the time and the criteria for being linked on the site feels different to, say, being on a page in the wiki)
1. The OSP homepage should fit both the "what" and the "why" -- but the "why" should be prioritized

**Reply by antistic** [reply]
Some more notes:

a. The plover site
  - it's hard to find the download links on the Plover page
    - People expect to see a download link that automatically detects their OS. (E.g. <https://discord.com/download>
    - Because the "download now" button is a different color, it looks "grayed out" and not clickable
    - Clicking the "download now" button doesn't actually download it now, it links to the GitHub release page which is not very accessible (the download links are at the bottom, and you have to have some technical knowledge to know which one to download)
    - The beginner's guide is hard to read (a lot of text, hard to follow, have to click a lot of links before you get to the download)
    - The Plover site looks outdated and not in line with the Open Steno Project
    - It might be good to have the steno explainer video on the Plover site too
    
b.  it might be a good idea to include a download link to Plover on the OSP site
c. OSP and Plover are not exactly the same but does a new user need to know about this difference if they just want to get started with Plover?

**Reply by antistic** [reply]
OSP marketing: "with it you can write over two hundred words per minute", but speed isn't really the thing we love steno for, and people think we should emphasize that it's still worth it even if you don't get to any faster than your normal typing speed. Instead, or in addition, we should advertise things like ergonomics and steno being fun.

**Reply by antistic** [reply]
separate setup guide for people who already know stenography

**Reply by panathea** [reply]
If the what is:

1. Community
2. Software
3. Education
4. Hardware

What is the why? From your comment it sounds like:

1. Fun
2. Speed
3. Ergonomics

Can you think of additional what's and why's, and would you order them differently by priority?

**Reply by antistic** [reply]
The "what" should also include what stenography is. I think that prioritisation looks good, although I would hesitate to put one over the other (they're all important!).

**Comment by user202729** [comment]
It's okay if you can work on it.

For the Open Steno Project and Plover homepage, the repository is at https://github.com/openstenoproject/openstenoproject.github.io and you can make a pull request.

For the wiki, it's already possible to edit it directly, you may want to consult the community (by hosting an alternative wiki for example?) before making a large reorganization.

RTD isn't that important, in comparison.

**Reply by antistic** [reply]
Yep, I'm willing to work on it. Part of what I'd like the conversation to be is to get an idea of what information people think is useful and where to put it. Nailing down a structure that will work for all use cases will take some time, I think.

RTD started off as an attempt to be the alternatively hosted wiki, and so far it's done well on the developer-oriented side of things, but not others.

Thanks for the link to the Open Steno Project homepage. The Plover homepage is actually in this repository, under the gh-pages branch, which I think contributes to it seeming disconnected from the main project.

**Reply by SeaLiteral** [reply]
Yeah, I've used the "wiki fork" (which isn't really a fork  since it isn't using the text from the existing wiki) to post a guide to Plover theory's vowels for speakers of languages that only have five vowels (such as Spanish). I wrote that guide some time ago, but I thought it would be better to have it somewhere where it was easier to find and editable by others. Although right now it's a user subpage, so it might look like it's not really meant to be edited much by others. At least linking to that thing is slightly more convenient than linking to a Discord post.

**Comment by antistic** [comment]
There is now a new wiki at <https://opensteno.miraheze.org/>. There is also a new discord thread at [#documentation-reorg](https://discord.com/channels/136953735426473984/935300573559939142) for the ongoing project of improving documentation.

- Why Miraheze?
  - We can easily export the data
  - We can easily import it somewhere else if we decide this isn't the right place (lots of converters to e.g. markdown exist)
  - It's free and open source (no need to fiddle around with hosting when we are experimenting)
- Why not a GitHub-backed markdown wiki (not completely ruled out for the future)?
  - PRs can take a while (could be solved by more people with PR approval permission and/or automating approvals)
  - Many people are not familiar with GitHub who would be more familiar with wikis
  - Going through "official open steno" limits the kind of content that is appropriate
  - Would need to decide what technology to use, who would implement it, etc. which would take a while
- Why not the existing GitHub wiki?
  - SEO is terrible
  - No easy way to search page contents
  - Attached to the Plover GitHub rather than open steno as a whole
